### ISSUE TITLE
Title bar has no visible focus state

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#51818]
### Description

The title bar, as currently implemented, is an important control to trigger the command center. It does not have any visible focus state, which is required to meet accessibility expectations.

### Step-by-step reproduction instructions

1. Open site editor.
2. Navigate with keyboard to locate the title bar.

Since there's no visible focus, you can use a tool like Paul Adam's [force show keyboard focus bookmarklet](https://pauljadam.com/bookmarklets/focus.html) to observe focus location.

### Environment info

- WordPress 6.3-alpha-55973
- Gutenberg 16.0.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
